applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_neighbouring_element_finder.cpp

```diff
@@ -0,0 +1,268 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_utilities/neighbouring_element_finder.hpp"
+#include "test_setup_utilities/element_setup_utilities.h"
+#include "test_setup_utilities/model_setup_utilities.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+namespace
+{
+using namespace Kratos;
+
+void CreateNumberOfNewNodes(ModelPart& rModelPart, std::size_t NumberOfNodes)
+{
+    for (std::size_t i = 0; i < NumberOfNodes; ++i) {
+        rModelPart.CreateNewNode(i + 1, 0.0, 0.0, 0.0);
+    }
+}
+
+PointerVector<Node> GetNodesFromIds(ModelPart& rModelPart, const std::vector<std::size_t>& rNodeIds)
+{
+    PointerVector<Node> result(rNodeIds.size());
+    std::ranges::transform(rNodeIds, result.ptr_begin(),
+                           [&rModelPart](auto Id) { return rModelPart.pGetNode(Id); });
+    return result;
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(NeighbouringElementFinder_ReturnsEmptyListWhenNoNeighbouringElements,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& r_model_part_for_neighbouring_elements = model.CreateModelPart("empty");
+
+    NeighbouringElementFinder finder;
+
+    auto& r_model_part_for_entities_for_finding =
+        ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model);
+
+    std::map<std::size_t, std::unique_ptr<BoundaryGenerator>> boundary_generators;
+    boundary_generators[std::size_t{2}] = std::make_unique<EdgesGenerator>();
+    finder.FindEntityNeighbours(r_model_part_for_entities_for_finding.Elements(),
+                                r_model_part_for_neighbouring_elements.Elements(), boundary_generators);
+
+    EXPECT_EQ(r_model_part_for_entities_for_finding.GetElement(1).GetValue(NEIGHBOUR_ELEMENTS).size(), 0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NeighbouringElementFinder_ReturnsCorrectNeighbouringElementOfElement,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+
+    auto& r_model_part_for_entities_for_finding =
+        ModelSetupUtilities::CreateModelPartWithASingle2D3NElement(model);
+
+    auto& r_model_part_for_neighbouring_elements = model.CreateModelPart("Main");
+
+    std::vector<std::size_t> neighbour_node_ids = {1, 2};
```

Comment:
This vector can be made `const`:
```suggestion
    const auto neighbour_node_ids = std::vector<std::size_t>{1, 2};
```
This suggestion also applies to the other unit tests.

Reply:
Done